lightly/transforms/random_crop_and_flip_with_grid.py

```diff
@@ -0,0 +1,147 @@
+    H: float
+    W: float
+    flip: bool = False
+
+
+@dataclass
+class ImageAndLocation:
+    image: Image.Image
+    location: Location
+
+
+@dataclass
+class ImageTensorAndGrid:
+    image: torch.Tensor
+    grid: torch.Tensor
+
+
+class RandomResizedCropWithLocation(T.RandomResizedCrop):
+    """
+    Do a random resized crop and return both the resulting image and the grid
+
+    """
+
+    def forward(self, img: ImageAndLocation) -> ImageAndLocation:
+        i, j, h, w = self.get_params(img.image, self.scale, self.ratio)
+        width, height = T.functional.get_image_size(img.image)
+        location = Location(i=i, j=j, h=h, w=w, H=height, W=width)
+        imageandlocation = ImageAndLocation(
+            image=T.functional.resized_crop(
+                img.image, i, j, h, w, self.size, self.interpolation
+            ),
+            location=location,
+        )
+        return imageandlocation
+        
+
+
+class RandomHorizontalFlipReturnsIfFlip(T.RandomHorizontalFlip):
+    """Horizontally flip the given image randomly with a given probability.
+    If the image is torch Tensor, it is expected
+    to have [..., H, W] shape, where ... means an arbitrary number of leading
+    dimensions
+
+
+    Args:
+        p (float): probability of the image being flipped. Default value is 0.5
+    """
+
+    def forward(self, img: ImageAndLocation) -> ImageAndLocation:
+        """
+        Args:
+            img (PIL Image or Tensor): Image to be flipped.
+        Returns:
+            PIL Image or Tensor: Randomly flipped image.
+        """
+        
+        if torch.rand(1) < self.p:
+            img.image = F.hflip(img.image)
+            img.location.flip = True
+        return img
+
+
+class ReturnGrid(nn.Module):
+    def __init__(
+        self,
+        N=7,
+        crop_size: int = 224,
+        crop_min_scale: float = 0.05,
+        crop_max_scale: float = 0.2,
+        hf_prob: float = 0.5,
+        normalize: dict = imagenet_normalize,
+    ):
+        super().__init__()
+        self.N = N
+        self.crop_size = crop_size
+        self.crop_min_scale = crop_min_scale
+        self.crop_max_scale = crop_max_scale
+        self.hf_prob = hf_prob
+        self.normalize = normalize
+
+    def forward(self, img: Image.Image) -> ImageTensorAndGrid:
+
+        location = Location(i=0, j=0, h=0, w=0, H=img.height, W=img.width)
+        img_and_location = ImageAndLocation(image=img, location=location)
+        img_and_location = RandomResizedCropWithLocation(
+            size=self.crop_size, scale=(self.crop_min_scale, self.crop_max_scale)
+        ).forward(img_and_location)
+
+        img_and_location = RandomHorizontalFlipReturnsIfFlip(self.hf_prob).forward(
+            img_and_location
+        )
+        transform = T.ToTensor()
+        image_tensor = transform(img_and_location.image)
+        image_tensor = T.Normalize(
+            mean=self.normalize["mean"], std=self.normalize["std"]
+        ).forward(image_tensor)
```

Comment:
Can you move all the created classes to be created in the init? This should make it faster.
In particular
- RandomResizedCropWithLocation
- RandomHorizontalFlipReturnsIfFlip
- T.ToTensor
- T.Normalize
The instance created of these classes are static. thus they can created in the init. Then you can reuse them in the forward pass.